### PR TITLE
refactor: StatSig flags that control individual features on onboarding name step

### PR DESCRIPTION
**Pull request by XrayBravoGolf**
### Description

Originally, a single flag on StatSig would control multiple parts of the `NameAndPicture` screen. 
This PR refactors the implementation, along with configurations inside StatSig, to have each flag control an individual feature on the screen that are part of experiment., such as input field placeholder, and whether to show or hide the button that generates a random username for the user.

By @satish-ravi :
> This gives us more flexibility in case we need to run more experiments in the future on this screen and reduces the complexity on the code, but does add complexity to the experiment configuration, and we also still need to honor existing flags (which we can clean up after current experiments are over)

### Other changes

Removed the legacy createAccountCopyTest code from `NameAndPicture.ts` and default to the applied result of experiment ("Create Profile").

### Tested

Automated testing are refactored to address additional flags.
Manual testing confirms placeholder, skip button and generator button works.

### How others should test

N/A

### Related issues

N/A

### Backwards compatibility

Compatible because the experiment has not started yet.

### CODE REVIEW

**Review by satish-ravi — COMMENTED** [review]
~Can you resolve conflicts and rebase with latest main?~

File: src/onboarding/registration/NameAndPicture.tsx [hunk 1]
```diff
@@ -46,11 +42,15 @@ const getExperimentParams = () => {
...
+    const showNameGeneratorButton = statsigLayer.get(
+      ExperimentParams[StatsigLayers.NAME_AND_PICTURE_SCREEN].showNameGeneratorButton.paramName,
+      ExperimentParams[StatsigLayers.NAME_AND_PICTURE_SCREEN].showNameGeneratorButton.defaultValue
     )
-    return [showSkipButton, nameType]
+    const namePlaceholder = statsigLayer.get(
+      ExperimentParams[StatsigLayers.NAME_AND_PICTURE_SCREEN].namePlaceholder.paramName,
+      ExperimentParams[StatsigLayers.NAME_AND_PICTURE_SCREEN].namePlaceholder.defaultValue
+    )
+    return [showSkipButton, showNameGeneratorButton, namePlaceholder]
```
[the remark this answers is not in the document]
Comment (XrayBravoGolf) [reply src/onboarding/registration/NameAndPicture.tsx:53]: Ooh yea, doing destructuring assignment on array really leaves no guard for this mismatch.

[the remark this answers is not in the document]
Comment (XrayBravoGolf) [reply src/onboarding/registration/NameAndPicture.tsx:53]: And this helped me catch a bug! 
6 lines below, the fail-safe in the catch block still used old default values.

File: src/app/types.ts [hunk 2]
```diff
@@ -17,8 +17,7 @@ export enum InviteMethodType {
   ReferralUrl = 'REFERRAL_URL',
 }
 
-export enum OnboardingNameType {
-  AutoGen = 'autogenerator',
-  FirstAndLast = 'first_and_last',
-  Placeholder = 'placeholder',
+export enum OnboardingNamePlaceholderType {
+  AlterEgo = 'myCryptoAlterEgo',
+  FullName = 'fullNamePlaceholder',
```
[the remark this answers is not in the document]
Comment (XrayBravoGolf) [reply src/app/types.ts:22]: Great idea!

**Comment by XrayBravoGolf** [comment]
Closed because the refactor didn't make into the version the experiment was expected to commence.

**Pull request closed without merging**